draft-gpew-priv-ppm.md

```diff
@@ -900,37 +900,70 @@ The named parameters are:
 * `batch_interval`, the request's batch interval.
 * `agg_param`, an aggregation parameter for the VDAF being executed.
 
-Depending on the VDAF scheme and how the leader is configured, the collect
-request may cause the leader to send a series of aggregate requests to the
-helpers in order to compute their share of the aggregate. Alternately, the
-leader may already have made these requests and can respond immediately. In
-either case it responds to the collector's request as follows.
+Depending on the VDAF scheme and how the leader is configured, the leader and
+helper may already have prepared all the reports falling within `batch_interval`
+and be ready to return the aggregate shares right away, but this cannot be
+guaranteed and in some VDAFs, it may not be possible to begin preparing inputs
+until the collector provides the aggregation parameter in the `CollectReq`. For
+those reasons, collect requests are handled asynchronously.
+
+Upon receipt of a `CollectReq`, the leader begins by checking that the request
+meets the requirements of the batch parameters using the procedure in
+{{batch-parameter-validation}}. If so, it immediately sends the collector a
+response with HTTP status 202 Accepted. The body consists of a `CollectResp`:
 
-It begins by checking that the request meets the requirements of the batch
-parameters using the procedure in {{batch-parameter-validation}}. If so, it
-obtains the helper's encrypted aggregate share for the batch interval by sending
-an AggregateShareReq to the helper as described in {{aggregate-share-request}}.
-(This request may too have been made in advance.)
-
-Next, the leader computes its own aggregate share by aggregating all of the
-verified output shares that fall within the batch interval. Finally, it responds
-with HTTP status 200 and a body consisting of a CollectResp message:
-
-[OPEN ISSUE: What happens if this all takes a really long time.]
+~~~
+struct {
+  Url collect_job;
+} CollectResp;
+~~~
 
-[TODO: Decide if and how the collector's request is authenticated.]
+`CollectResp.collect_job` is a URL identifying the collect job that can be
+polled by the collector. It SHOULD contain an unpredictable component to
+prevent enumeration attacks against a leader's collect results. The HTTP
+response MAY include a Retry-After header field to suggest a polling interval to
+the client.
+
+The leader then begins working with the helper to prepare the shares falling
+into `CollectReq.batch_interval` (or continues this process, depending on the
+VDAF) as described in {{pa-aggregate}}. After receiving a `CollectResp`, the
+collector periodically makes HTTP GET requests to `CollectResp.collect_job` to
+check on the status of the collect job and eventually obtain the result. If the
+collect job is not finished yet, the leader responds with HTTP status 202
+Accepted and a `CollectResp`. The `CollectResp` MAY contain a different
+`collect_job` than previous `CollectResp`s and the HTTP response MAY include a
```

Comment:
Q: what's the practical reason for allowing the `collect_job` to change?

Reply:
I didn't see a reason to require that it be the same, but I suppose we could instead insist that it always be the same. Or we could say nothing, implicitly permitting `collect_job` to change.

Reply:
WDYT about having the `collect_job` handler return an HTTP 202 with an unspecified (or empty) body if the results aren't yet available? This removes the unnecessary flexibility of changing `collect_jobs` without requiring specification language.

This is pretty nit-picky; I'd personally go with an unspecified body to leave things maximally flexible while allowing us to specify later, but LGTM either way.

Reply:
Allowing the collect job URL to change seems to add some complexity. If we don't need this complexity, than I suggest we remove it. In particular,  I'd propose removing this optional behavior.

Reply:
I prefer @BranLwyd's suggestion here. Note that otherwise we have weird race conditions.

Reply:
I will incorporate Bran's idea.